engine/src/main/scala/cromwell/webservice/WorkflowSubmitBatchActor.scala

```diff
@@ -0,0 +1,47 @@
+package cromwell.webservice
+
+import akka.actor.{Actor, ActorRef, Props}
+import cromwell.engine.WorkflowSourceFiles
+import cromwell.engine.workflow.WorkflowManagerActor
+import cromwell.webservice.CromwellApiHandler.{WorkflowManagerBatchSubmitResponse, WorkflowManagerResponse}
+
+/** Submits a sequence of sources, then messages back the respective sequence of responses. */
+class WorkflowSubmitBatchActor(responseActor: ActorRef, requestHandlerActor: ActorRef,
+                               sources: Seq[WorkflowSourceFiles]) extends Actor {
+
+  private val responses = Array.fill[Option[WorkflowManagerResponse]](sources.size)(None)
+
+  override def preStart() = {
+    sources.zipWithIndex foreach {
+      case (source, index) =>
+        context.actorOf(
+          Props(new WorkflowSubmitIndexedActor(self, requestHandlerActor, source, index)),
+          "WorkflowSubmitIndexedActor_" + index)
+    }
+  }
+
+  override def receive = {
+    case WorkflowManagerBatchIndexedResponse(response, index) =>
+      responses(index) = Option(response)
```

Comment:
No change required (it would be massive overkill)... I just want to mention a slight sense of sadness that this isn't an FSM :)